mlir/include/mlir/Dialect/SCF/Utils/Utils.h

```diff
@@ -100,11 +100,16 @@ getSCFMinMaxExpr(Value value, SmallVectorImpl<Value> &dims,
 /// `loops` contains a list of perfectly nested loops with bounds and steps
 /// independent of any loop induction variable involved in the nest.
 LogicalResult coalesceLoops(MutableArrayRef<scf::ForOp> loops);
```

Comment:
Should we remove the version that doesn't take the rewriter?

Reply:
I am fine doing that. Should be a simple fix for downstream users.

Reply:
Actually tried it, but this has some strange coupling with things in Affine dialect. I am just going to leave it there for now for people to use at their own risk.